src/Components/List/List.jsx

```diff
@@ -38,18 +71,60 @@ const purchaseItem = (item, token) => {
         lastPurchased: new Date(),
         oldPurchased: item.lastPurchased,
         numberOfPurchases: numberOfPurchases,
-        calculatedEstimate: calculateEstimate(
-          item.frequency,
-          lastInterval,
-          numberOfPurchases,
-        ),
+        calculatedEstimate: calculatedEstimate,
       },
     });
 };
 
+//Color code based on days until next purchase or inactivity with corresponding aria label
+const colorCode = (item) => {
+  const daysSincePurchased = dayjs().diff(
+    getLastPurchaseDate(item, item.lastPurchased),
+    'day',
+  );
+  const estimatedCountdown = item.calculatedEstimate - daysSincePurchased;
+  const elapsedTime = dayjs().isSameOrAfter(
+    dayjs(getLastPurchaseDate(item, item.lastPurchased)).add(
+      item.calculatedEstimate * 2,
+      'day',
+    ),
+  );
+
+  if (elapsedTime === true) {
+    return ['inactive', 'Inactive'];
+  } else if (estimatedCountdown <= 7) {
+    return ['soon', 'Within 7 days'];
+  } else if (estimatedCountdown <= 14) {
+    return ['kindOfSoon', 'Between 7 and 14 days'];
+  } else if (14 < estimatedCountdown) {
+    return ['notSoon', 'More than 14 days'];
+  } else if (isNaN(estimatedCountdown)) {
+    return ['notBought', 'Never Bought Yet'];
+  } else {
+    return null;
+  }
+};
+
+// Sort items by soonest to latest estimated repurchase
+const sortItems = (a, b) => {
+  if (
+    a.calculatedEstimate < b.calculatedEstimate ||
+    b.calculatedEstimate === undefined
+  ) {
+    return -1;
+  } else if (a.calculatedEstimate > b.calculatedEstimate) {
+    return 1;
+  } else if (a.calculatedEstimate === b.calculatedEstimate) {
+    //This doesn't order names as it should. Tried localeCompare, > <, -, nested if
+    return a.name.localeCompare(b.name, 'en', { sensitivity: 'base' });
+  }
+  console.log(a.name);
+};
```

Comment:
I think I figured why the items that have `undefined` for their `calculatedEstimate` weren't being sorted alphabetically. 

When both items have an `undefined` calculated estimate, it triggers line 112 which will always cause it to return -1 and it never reaches line 119 to sort alphabetically. One solution to get around this is to move lines 117-120 to the beginning of the if statement, so that it checks if both items have the same `calculatedEstimate` first. Hope that helps!

Reply:
This is why fresh eyes are always a good thing! Such a simple fix that caused so many issues😅. That worked and has been pushed! Thanks a bunch🙏!